fenix/app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -1166,9 +1166,33 @@ class Settings(private val appContext: Context) : PreferencesHolder {
             return false
         }
 
-    var openLinksInExternalApp by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_open_links_in_external_app),
-        default = false,
+    /**
+     * Check to see if we should open the link in an external app
+     */
+    fun shouldOpenLinksInApp(): Boolean {
+        return when (openLinksInExternalApp) {
+            appContext.getString(R.string.app_links_ask_before_opening_string) -> true
+            appContext.getString(R.string.app_links_always_string) -> true
+            appContext.getString(R.string.app_links_never_string) -> false
+            else -> false
+        }
+    }
+
+    /**
+     * Check to see if we need to prompt the user if the link can be opened in an external app
+     */
+    fun shouldPromptOpenLinksInApp(): Boolean {
+        return when (openLinksInExternalApp) {
+            appContext.getString(R.string.app_links_ask_before_opening_string) -> true
+            appContext.getString(R.string.app_links_always_string) -> false
+            appContext.getString(R.string.app_links_never_string) -> true
+            else -> false
```

Comment:
I wonder if this else should return true

Reply:
Agreed.  Even though we're not expecting that case to happen, we should default to true.  I'll update.  Thanks